workflow/controller/operator.go

```diff
@@ -316,6 +322,18 @@ func (woc *wfOperationCtx) persistUpdates() {
 			return
 		}
 	}
+
+	if db != nil && !db.IsInterfaceNil() {
```

Comment:
I don't understand this check. db.IsInterfaceNil() is doing the same check (db == nil). It seems redundant.